packages/devextreme/js/__internal/core/license/license_validation.ts

```diff
@@ -78,3 +82,53 @@ export function parseToken(encodedToken: string | undefined): Token {
     },
   };
 }
+
+export type LicenseVerifyResult = 'W0019' | 'W0020' | 'W0021' | null;
+let isLicenseVerified = false;
+
+const verifyLicense = function verifyLicense(licenseToken: string, ver: string = version): void {
+  if (isLicenseVerified) {
+    return;
+  }
+  isLicenseVerified = true;
+
+  let warning: LicenseVerifyResult = null;
+
+  if (licenseToken) {
+    const license = parseToken(licenseToken);
+
+    if (license.kind === CORRUPTED) {
+      warning = 'W0021';
+    } else {
+      const [major, minor] = ver.split('.').map(Number);
+
+      if (major * 10 + minor > license.payload.maxVersionAllowed) {
+        warning = 'W0020';
+      }
+    }
+  } else {
+    warning = 'W0019';
+  }
+
+  if (warning) {
+    errors.log(warning);
+  }
+};
```

Comment:
Minor comment.
(You can apply it or not by your opinion)

I suggest to split this function into two.
Something like this:
```suggestion
const verifyLicense = (licenseToken: string, version: string = packageVersion): void {
  if (isLicenseVerified) {
    return;
  }
  isLicenseVerified = true;

  const [success, warning] = parseLicense(licenseToken, version);
  
  if (!success) {
    errors.log(warning);
  }
}

const parseLicense = (
  licenseToken: string,
  ver: string = versionValue,
): [success: boolean, warning: string | null] {
  if (!licenseToken) {
    return [false, 'W0019'];
  }

  const license = parseToken(licenseToken);

  if (license.kind === TokenKind.currupted) {
    return [false, 'W0021'];
  }

  /* See additional comment for this part */
  const [major, minor] = ver.split('.').map(Number);

  if (major * 10 + minor > license.payload.maxVersionAllowed) {
    return [false, 'W0020'];
  }
  /* end of this part :) */

  return [true, null];
}
```

In my opinion, it makes this code clearer and more open to modification in future